### ISSUE TITLE
Can't publish .net Core Console app - referencing Shared Project with conditional build symbols

### ISSUE THREAD

**Issue by Pinox** [dotnet/sdk#1124]
Having problems publishing a .net core console app where I simply reference a Shared project using a conditional build symbol.

In shared project I have the following reference that gives me an error.

#if (!CORE_APP)
using SQLite.Net.Attributes;
#endif

This is my dotnet publish error: "The type or namespace name 'SQLite' could not be found (are you missing a using directive or an assembly reference?)"

The above runs perfectly in VS2017 but I can't seem to publish the app. It picks up the reference that it shouldn't

Sample hello app that replicates error:
https://drive.google.com/open?id=0B0Gm-m-z_U84U1Z5XzFValNjaFU

Any suggestions would be appreciated.

**Comment by livarcocc** [comment]
Which version of the CLI are you using? I just tried your repro project using CLI 1.0.1 and I could restore and build, publish and run Console_CheckSharedLib.

**Comment by Pinox** [comment]
I get the error in CLI version 1.0.2 and 1.0.3

**Comment by livarcocc** [comment]
can you do a dotnet build /v:diag and share the log file? I have tried it a couple of times locally and I can successfully build and run the app. Though, to be fair, I tried on OSX, but I don't expect that to make any difference.

I wonder if you maybe have a CONFIG environment variable set, which may be causing this. I think I remember seeing that some machines, like some HP boxes, have that and other env variables that interfere with MSBuild set.

**Comment by Pinox** [comment]
thanks @livarcocc for pointing me in the right direction  - it seems my pain finally ended ;)

After deleting and reinstalling the CLI SDK's I ended up with a different error on the above sample project.

The new error on the project was now "configuration Debug|MCD is invalid".  After searching the internet and taking into account your statement of environment variables I ended up deleting the platform variable from the registry.

Open registry
Search Path
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment
Delete "Platform" variable
Reboot system

In my case the platform value was MCD and deleting it fix all my problems, I can now build and publish .net core apps again.

As per your suspicion, I am running the apps on a HP laptop.

Thanks again.
